frontend/src/components/Registration.js

```diff
@@ -0,0 +1,135 @@
+              password: '',
+              kingdomname: ''
+            }
+          };
+          this.handleSubmit = this.handleSubmit.bind(this);
+          this.handleUsername = this.handleUsername.bind(this);
+          this.handlePassword = this.handlePassword.bind(this);
+          this.handleKingdomname = this.handleKingdomname.bind(this);
+        }
+
+
+        handleSubmit(event) {
+
+          event.preventDefault();
+
+            let userData = this.state.newUser;
+            //let history = useHistory();
+            const { history } = this.props;
+
+            fetch('http://localhost:5000/api/users',{
+                method: "POST",
+                body: JSON.stringify(userData),
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json'
+                },
+              }).then(response => 
+                response.json()
+                ).then(data => {
+                  
+                  if(data.error){
+                    alert(data.error)
+                  }else{
+                    alert("welcome " + data.username)
+                   history.push('/login');
+                  }
+                }   
+              )
+        }
+
+        handleUsername(e){
+          let value = e.target.value;
+          this.setState( prevState => ({ newUser :
+            {...prevState.newUser, username: value
+            }
+          }))
+        }
+
+        handlePassword(e){
+          let value = e.target.value;
+          this.setState( prevState => ({ newUser :
+            {...prevState.newUser, password: value
+            }
+          }))
+        }
+
+        handleKingdomname(e){
+          let value = e.target.value;
+          this.setState( prevState => ({ newUser :
+            {...prevState.newUser, kingdomname: value
+            }
+          }))
+        }
+        
+        validUsername(){
+          return (this.state.newUser.username.length > 0)
+        }
+
+        validPassword(){
+          return !(String(this.state.newUser.password).length < 8);
+        }
+
+
+        validKingdomname(){
+          return !(String(this.state.newUser.kingdomname).length < 1)
+        }
+        
+
+        render() {
+
+            return (<div className="container">
+        
+            <div className="title">
+            <h2>Tribes of Vulpes</h2>
+            </div>
+
+            <div className="form">
+                <form 
+               
+                onSubmit = {this.handleSubmit}>
+                    <div>
+                    <input type="text" name="username" className={this.validUsername() ? "green" : "red"} placeholder="Username" onChange = {this.handleUsername}>
+                    </input>
+                    <div className={this.validUsername() ? "" : "redimage"}><span className={this.validUsername() ? "alertnotext" : ""}>Username must be at least 1 character long!</span></div>
+                    </div>

```

Comment:
@zvorsi if it's valid you can just leave out the whole div (and the same goes for the others)
```suggestion
                    {!this.validUsername() && <div className="redimage"><span>Username must be at least 1 character long!</span></div>}
                    </div>

```